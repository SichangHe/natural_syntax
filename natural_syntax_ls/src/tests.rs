use super::*;

use insta::assert_debug_snapshot;
use tracing::Level;
use tracing_subscriber::EnvFilter;

#[test]
fn token_types() {
    init_tracing();
    let types = semantic_token_types();
    assert_debug_snapshot!(types);
}

#[test]
fn token_modifiers() {
    init_tracing();
    let modifiers = semantic_token_modifiers();
    assert_debug_snapshot!(modifiers);
}

fn init_tracing() {
    _ = tracing_subscriber::fmt()
        .with_env_filter(
            EnvFilter::builder()
                .with_default_directive(Level::INFO.into())
                .from_env_lossy(),
        )
        .try_init();
}
