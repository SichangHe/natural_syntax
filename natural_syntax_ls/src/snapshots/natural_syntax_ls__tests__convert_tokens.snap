---
source: natural_syntax_ls/src/tests.rs
expression: combined
---
[
    (
        POSToken {
            word: "The",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 1,
            word_index: 1,
            offset_begin: 0,
            offset_end: 3,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 0,
            length: 3,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "easiest",
            score: 0.8220000000000001,
            tag: JJS,
            label_index: 9,
            index: 2,
            word_index: 2,
            offset_begin: 4,
            offset_end: 11,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 512,
        },
    ),
    (
        POSToken {
            word: "way",
            score: 0.998,
            tag: NN,
            label_index: 11,
            index: 3,
            word_index: 3,
            offset_begin: 12,
            offset_end: 15,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 3,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "of",
            score: 1.0,
            tag: IN,
            label_index: 6,
            index: 4,
            word_index: 4,
            offset_begin: 16,
            offset_end: 18,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "getting",
            score: 0.996,
            tag: VBG,
            label_index: 27,
            index: 5,
            word_index: 5,
            offset_begin: 19,
            offset_end: 26,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 7,
            token_type: 12,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "started",
            score: 0.8320000000000001,
            tag: VBN,
            label_index: 28,
            index: 6,
            word_index: 6,
            offset_begin: 27,
            offset_end: 34,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 7,
            token_type: 12,
            token_modifiers_bitset: 512,
        },
    ),
    (
        POSToken {
            word: "is",
            score: 1.0,
            tag: VBZ,
            label_index: 30,
            index: 7,
            word_index: 7,
            offset_begin: 35,
            offset_end: 37,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 2,
            token_type: 12,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "the",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 8,
            word_index: 8,
            offset_begin: 38,
            offset_end: 41,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 3,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "rustler",
            score: 0.364,
            tag: NNP,
            label_index: 12,
            index: 10,
            word_index: 10,
            offset_begin: 43,
            offset_end: 50,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "library",
            score: 0.6910000000000001,
            tag: NN,
            label_index: 11,
            index: 15,
            word_index: 12,
            offset_begin: 58,
            offset_end: 65,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 15,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "https",
            score: 0.468,
            tag: NNS,
            label_index: 14,
            index: 18,
            word_index: 15,
            offset_begin: 67,
            offset_end: 72,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 9,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "hex",
            score: 0.805,
            tag: NNP,
            label_index: 12,
            index: 22,
            word_index: 19,
            offset_begin: 75,
            offset_end: 78,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 3,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "pm",
            score: 0.973,
            tag: NNP,
            label_index: 12,
            index: 25,
            word_index: 21,
            offset_begin: 79,
            offset_end: 81,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 2,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "packages",
            score: 0.646,
            tag: NNS,
            label_index: 14,
            index: 27,
            word_index: 23,
            offset_begin: 82,
            offset_end: 90,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 8,
            token_type: 1,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "Add",
            score: 0.926,
            tag: VB,
            label_index: 25,
            index: 34,
            word_index: 29,
            offset_begin: 104,
            offset_end: 107,
        },
        SemanticToken {
            delta_line: 2,
            delta_start: 2,
            length: 3,
            token_type: 12,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "the",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 35,
            word_index: 30,
            offset_begin: 108,
            offset_end: 111,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 3,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "library",
            score: 0.66,
            tag: NN,
            label_index: 11,
            index: 42,
            word_index: 34,
            offset_begin: 128,
            offset_end: 135,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 20,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "https",
            score: 0.47700000000000004,
            tag: NNS,
            label_index: 14,
            index: 45,
            word_index: 37,
            offset_begin: 137,
            offset_end: 142,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 9,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "hex",
            score: 0.801,
            tag: NNP,
            label_index: 12,
            index: 49,
            word_index: 41,
            offset_begin: 145,
            offset_end: 148,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 3,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "pm",
            score: 0.963,
            tag: NNP,
            label_index: 12,
            index: 52,
            word_index: 43,
            offset_begin: 149,
            offset_end: 151,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 2,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "packages",
            score: 0.605,
            tag: NNS,
            label_index: 14,
            index: 54,
            word_index: 45,
            offset_begin: 152,
            offset_end: 160,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 8,
            token_type: 1,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "as",
            score: 0.994,
            tag: IN,
            label_index: 6,
            index: 59,
            word_index: 49,
            offset_begin: 170,
            offset_end: 172,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 18,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "a",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 60,
            word_index: 50,
            offset_begin: 173,
            offset_end: 174,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 1,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "dependency",
            score: 0.998,
            tag: NN,
            label_index: 11,
            index: 61,
            word_index: 51,
            offset_begin: 177,
            offset_end: 187,
        },
        SemanticToken {
            delta_line: 1,
            delta_start: 2,
            length: 10,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "of",
            score: 1.0,
            tag: IN,
            label_index: 6,
            index: 62,
            word_index: 52,
            offset_begin: 188,
            offset_end: 190,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 11,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "your",
            score: 0.786,
            tag: PRP,
            label_index: 17,
            index: 63,
            word_index: 53,
            offset_begin: 191,
            offset_end: 195,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 4,
            token_type: 6,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "project",
            score: 0.994,
            tag: NN,
            label_index: 11,
            index: 64,
            word_index: 54,
            offset_begin: 196,
            offset_end: 203,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "Run",
            score: 0.9590000000000001,
            tag: VB,
            label_index: 25,
            index: 67,
            word_index: 57,
            offset_begin: 207,
            offset_end: 210,
        },
        SemanticToken {
            delta_line: 1,
            delta_start: 2,
            length: 3,
            token_type: 12,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "mix",
            score: 0.794,
            tag: VB,
            label_index: 25,
            index: 69,
            word_index: 59,
            offset_begin: 212,
            offset_end: 215,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 3,
            token_type: 12,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "new",
            score: 0.525,
            tag: JJ,
            label_index: 7,
            index: 73,
            word_index: 62,
            offset_begin: 224,
            offset_end: 227,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 12,
            length: 3,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "to",
            score: 1.0,
            tag: TO,
            label_index: 23,
            index: 75,
            word_index: 64,
            offset_begin: 229,
            offset_end: 231,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "generate",
            score: 0.99,
            tag: VB,
            label_index: 25,
            index: 76,
            word_index: 65,
            offset_begin: 232,
            offset_end: 240,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 8,
            token_type: 12,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "a",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 77,
            word_index: 66,
            offset_begin: 241,
            offset_end: 242,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 9,
            length: 1,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "new",
            score: 0.997,
            tag: JJ,
            label_index: 7,
            index: 78,
            word_index: 67,
            offset_begin: 243,
            offset_end: 246,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 2,
            length: 3,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "NIF",
            score: 0.41400000000000003,
            tag: NN,
            label_index: 11,
            index: 79,
            word_index: 68,
            offset_begin: 247,
            offset_end: 250,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 3,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "in",
            score: 1.0,
            tag: IN,
            label_index: 6,
            index: 81,
            word_index: 69,
            offset_begin: 251,
            offset_end: 253,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "your",
            score: 0.608,
            tag: PRP,
            label_index: 17,
            index: 82,
            word_index: 70,
            offset_begin: 254,
            offset_end: 258,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 4,
            token_type: 6,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "project",
            score: 0.997,
            tag: NN,
            label_index: 11,
            index: 83,
            word_index: 71,
            offset_begin: 259,
            offset_end: 266,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "Follow",
            score: 0.977,
            tag: VB,
            label_index: 25,
            index: 85,
            word_index: 73,
            offset_begin: 268,
            offset_end: 274,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 9,
            length: 6,
            token_type: 12,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "the",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 86,
            word_index: 74,
            offset_begin: 275,
            offset_end: 278,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 7,
            length: 3,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "instructions",
            score: 0.995,
            tag: NNS,
            label_index: 14,
            index: 87,
            word_index: 75,
            offset_begin: 281,
            offset_end: 293,
        },
        SemanticToken {
            delta_line: 1,
            delta_start: 2,
            length: 12,
            token_type: 1,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "If",
            score: 0.991,
            tag: IN,
            label_index: 6,
            index: 90,
            word_index: 78,
            offset_begin: 297,
            offset_end: 299,
        },
        SemanticToken {
            delta_line: 1,
            delta_start: 2,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "you",
            score: 0.974,
            tag: PRP,
            label_index: 17,
            index: 91,
            word_index: 79,
            offset_begin: 300,
            offset_end: 303,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 3,
            token_type: 6,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "are",
            score: 0.934,
            tag: VBP,
            label_index: 29,
            index: 92,
            word_index: 80,
            offset_begin: 304,
            offset_end: 307,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 3,
            token_type: 12,
            token_modifiers_bitset: 4,
        },
    ),
    (
        POSToken {
            word: "already",
            score: 0.996,
            tag: RB,
            label_index: 18,
            index: 93,
            word_index: 81,
            offset_begin: 308,
            offset_end: 315,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 7,
            token_type: 15,
            token_modifiers_bitset: 128,
        },
    ),
    (
        POSToken {
            word: "using",
            score: 0.998,
            tag: VBG,
            label_index: 27,
            index: 94,
            word_index: 82,
            offset_begin: 316,
            offset_end: 321,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 5,
            token_type: 12,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "serde",
            score: 0.547,
            tag: NNP,
            label_index: 12,
            index: 97,
            word_index: 85,
            offset_begin: 324,
            offset_end: 329,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "https",
            score: 0.48,
            tag: NNS,
            label_index: 14,
            index: 102,
            word_index: 89,
            offset_begin: 332,
            offset_end: 337,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 8,
        },
    ),
    (
        POSToken {
            word: "serde",
            score: 0.79,
            tag: NNP,
            label_index: 12,
            index: 106,
            word_index: 93,
            offset_begin: 340,
            offset_end: 345,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "rs",
            score: 0.672,
            tag: NNP,
            label_index: 12,
            index: 109,
            word_index: 95,
            offset_begin: 346,
            offset_end: 348,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 6,
            length: 2,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "and",
            score: 1.0,
            tag: CC,
            label_index: 1,
            index: 111,
            word_index: 97,
            offset_begin: 350,
            offset_end: 353,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 3,
            token_type: 15,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "or",
            score: 0.985,
            tag: CC,
            label_index: 1,
            index: 113,
            word_index: 99,
            offset_begin: 354,
            offset_end: 356,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 4,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "have",
            score: 0.556,
            tag: VBP,
            label_index: 29,
            index: 114,
            word_index: 100,
            offset_begin: 357,
            offset_end: 361,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 4,
            token_type: 12,
            token_modifiers_bitset: 4,
        },
    ),
    (
        POSToken {
            word: "been",
            score: 0.999,
            tag: VBN,
            label_index: 28,
            index: 115,
            word_index: 101,
            offset_begin: 362,
            offset_end: 366,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 4,
            token_type: 12,
            token_modifiers_bitset: 512,
        },
    ),
    (
        POSToken {
            word: "using",
            score: 0.993,
            tag: VBG,
            label_index: 27,
            index: 116,
            word_index: 102,
            offset_begin: 367,
            offset_end: 372,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 5,
            token_type: 12,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "serde",
            score: 0.733,
            tag: NNP,
            label_index: 12,
            index: 118,
            word_index: 104,
            offset_begin: 376,
            offset_end: 381,
        },
        SemanticToken {
            delta_line: 1,
            delta_start: 3,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "rustler",
            score: 0.368,
            tag: NNP,
            label_index: 12,
            index: 121,
            word_index: 106,
            offset_begin: 382,
            offset_end: 389,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 6,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "before",
            score: 0.927,
            tag: IN,
            label_index: 6,
            index: 124,
            word_index: 108,
            offset_begin: 391,
            offset_end: 397,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 9,
            length: 6,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "please",
            score: 0.6980000000000001,
            tag: VB,
            label_index: 25,
            index: 126,
            word_index: 110,
            offset_begin: 399,
            offset_end: 405,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 6,
            token_type: 12,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "enable",
            score: 0.9540000000000001,
            tag: VB,
            label_index: 25,
            index: 127,
            word_index: 111,
            offset_begin: 406,
            offset_end: 412,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 7,
            length: 6,
            token_type: 12,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "the",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 128,
            word_index: 112,
            offset_begin: 413,
            offset_end: 416,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 7,
            length: 3,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "serde",
            score: 0.534,
            tag: NNP,
            label_index: 12,
            index: 130,
            word_index: 114,
            offset_begin: 418,
            offset_end: 423,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 5,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 1,
        },
    ),
    (
        POSToken {
            word: "feature",
            score: 0.982,
            tag: NN,
            label_index: 11,
            index: 133,
            word_index: 116,
            offset_begin: 425,
            offset_end: 432,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 7,
            length: 7,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "in",
            score: 1.0,
            tag: IN,
            label_index: 6,
            index: 134,
            word_index: 117,
            offset_begin: 433,
            offset_end: 435,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 8,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "your",
            score: 0.787,
            tag: PRP,
            label_index: 17,
            index: 135,
            word_index: 118,
            offset_begin: 436,
            offset_end: 440,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 4,
            token_type: 6,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "crate",
            score: 0.971,
            tag: NN,
            label_index: 11,
            index: 138,
            word_index: 120,
            offset_begin: 445,
            offset_end: 450,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 9,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "s",
            score: 0.675,
            tag: DT,
            label_index: 3,
            index: 140,
            word_index: 122,
            offset_begin: 451,
            offset_end: 452,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 6,
            length: 1,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "Cargo",
            score: 0.5690000000000001,
            tag: NN,
            label_index: 11,
            index: 142,
            word_index: 124,
            offset_begin: 456,
            offset_end: 461,
        },
        SemanticToken {
            delta_line: 1,
            delta_start: 3,
            length: 5,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
    (
        POSToken {
            word: "on",
            score: 0.996,
            tag: IN,
            label_index: 6,
            index: 147,
            word_index: 128,
            offset_begin: 468,
            offset_end: 470,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 12,
            length: 2,
            token_type: 15,
            token_modifiers_bitset: 64,
        },
    ),
    (
        POSToken {
            word: "the",
            score: 1.0,
            tag: DT,
            label_index: 3,
            index: 148,
            word_index: 129,
            offset_begin: 471,
            offset_end: 474,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 3,
            length: 3,
            token_type: 15,
            token_modifiers_bitset: 256,
        },
    ),
    (
        POSToken {
            word: "dependency",
            score: 0.997,
            tag: NN,
            label_index: 11,
            index: 153,
            word_index: 133,
            offset_begin: 485,
            offset_end: 495,
        },
        SemanticToken {
            delta_line: 0,
            delta_start: 14,
            length: 10,
            token_type: 1,
            token_modifiers_bitset: 0,
        },
    ),
]
